dask/dataframe/core.py

```diff
@@ -351,11 +351,22 @@ def cache(self, cache=Cache):
 
     @derived_from(pd.DataFrame)
     def drop_duplicates(self, split_every=None, split_out=1, **kwargs):
-        assert all(k in ('keep', 'subset', 'take_last') for k in kwargs)
+        # Let pandas error on bad inputs
+        self._meta_nonempty.drop_duplicates(**kwargs)
+        if 'subset' in kwargs and kwargs['subset'] is not None:
+            split_out_setup = split_out_on_cols
+            split_out_setup_kwargs = {'cols': kwargs['subset']}
+        else:
+            split_out_setup = split_out_setup_kwargs = None
+
+        if 'keep' in kwargs and kwargs['keep'] is False:
+            raise NotImplementedError("drop_duplicates with keep=False")
+
         chunk = M.drop_duplicates
         return aca(self, chunk=chunk, aggregate=chunk, meta=self._meta,
                    token='drop-duplicates', split_every=split_every,
-                   split_out=split_out, split_index=False, **kwargs)
+                   split_out=split_out, split_out_setup=split_out_setup,
+                   split_out_setup_kwargs=split_out_setup_kwargs, **kwargs)
```

Comment:
It is a bit verbose, but I doubt it'll be used that much. I think I prefer the clarity over the need to type a few more characters.